crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -0,0 +1,520 @@
+                        builder = builder.with_prop(
+                            azdls::ConfigKeys::AuthorityHost,
+                            authority_host.to_string(),
+                        );
+                    }
+                }
+            }
+        }
+
+        builder.build().map_err(|e| {
+            ErrorModel::precondition_failed(
+                "Error creating S3 filesystem.",
+                "S3FileIOError",
+                Some(Box::new(e)),
+            )
+            .into()
+        })
+    }
+
+    async fn validate_sas(
+        &mut self,
+        credential: Option<&AzCredential>,
+        ns_id: NamespaceIdentUuid,
+        table_id: Uuid,
+        test_location: String,
+    ) -> Result<()> {
+        let table_config = self
+            .generate_table_config(
+                Uuid::now_v7().into(),
+                table_id.into(),
+                ns_id.into(),
+                &DataAccess {
+                    vended_credentials: false,
+                    remote_signing: false,
+                },
+                credential.unwrap(),
+            )
+            .await
+            .map_err(|e| {
+                e.error
+                    .append_detail("Couldn't generate table config while validating.")
+            })?;
+
+        // TODO: replace with iceberg_rust's FileIO + opendal once sas is available
+        let cred = azure_storage::StorageCredentials::sas_token(
+            table_config
+                .get(self.iceberg_sas_property_key().as_str())
+                .ok_or(ErrorModel::internal(
+                    "Couldn't find sas token in table config.",
+                    "AzStorageSasTokenNotFound",
+                    None,
+                ))?,
+        )
+        .map_err(|e| {
+            ErrorModel::internal(
+                "Couldn't create storage credentials from sas token.",
+                "AzStorageSasTokenCreationError",
+                Some(Box::new(e)),
+            )
+        })?;
+        let client =
+            azure_storage_blobs::prelude::BlobServiceClient::new(self.account_name.as_str(), cred);
+        let container = client.container_client(self.filesystem.as_str());
+        let blob = container.blob_client(
+            // Blob client expects the path without protocol
+            super::path_utils::sanitize_azdls_path(test_location.as_str())
+                .trim_start_matches("abfss://"),
+        );
+        let mut get = blob.get().into_stream();
+        while let Some(n) = get.next().await {
+            if let Err(e) = n {
+                return Err(ErrorModel::internal(
+                    "Failed to read test location.",
+                    "AzStorageReadError",
+                    Some(Box::new(e)),
+                )
+                .into());
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn get_sas_token(&self, cred: StorageCredentials) -> Result<String> {
```

Comment:
We need to add the Table / View Location here. The token is currently too wide.

Reply:
changed to be more narrow, now we're taking location here